src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -847,11 +847,10 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 						reflectionContext.AnalyzingPattern ();
 
 						var parameters = calledMethod.Parameters;
-						BindingFlags bindingFlags = BindingFlags.Public | BindingFlags.Instance;
-						if (parameters.Count > 1) {
-							if (methodParams[1].AsConstInt () != null)
-								bindingFlags |= (BindingFlags) methodParams[1].AsConstInt ();
-						}
+						BindingFlags? bindingFlags = BindingFlags.Public | BindingFlags.Instance;
+						if (parameters.Count > 1 && calledMethod.Parameters[0].ParameterType.Name == "BindingFlags")
+							bindingFlags = GetBindingFlags (methodParams[1]);
```

Comment:
I think it'd be nicer to write such initialisations as if/else. Following is easier for me to read than the double initialization

```c#
	BindingFlags? bindingFlags;
	if (parameters.Count > 1 && calledMethod.Parameters[0].ParameterType.Name == "BindingFlags")
		bindingFlags = GetBindingFlags (methodParams[1]);
	else
		bindingFlags = BindingFlags.Public | BindingFlags.Instance;

```